### DISCUSSION TITLE
SaaS Application: scaling & sizing

### DISCUSSION THREAD

**Discussion by v-kessler** [ory/kratos#3039] [Q&A] [answered]
Our use case is to build a SaaS App for million of tenants/orgs. I understood Ory Kratos can scale and point to a different databases. But if understand that correctly then each kratos process/pod can point to only 1 db and its not parametrised. With that assumption is scenario 1 only possible or there is a way to get scenario 2 work as well. 

In scenario  one will require a dedicated pods of kratos and hydra for each customer. From hardware retirements 
- a single kratos pod need 256 MB.   
- single hydra pod needs ? MB. Assumption  128 MB
- all in HA need at  least to double pods per customer

Assumption per customer I need to provision: 2 kratos + 2 hydra pods = 2 *256 + 2*126 = 768 MB
Assumption we have 10000 customers : 768 MB * 10.000 = 7680000 MB = 7500GB which need roughly 60 nodes of 128 GB RAM.

Are my conclusions correct:
- only scenario 1 is doable or scenario 2 can be implemented as well
- By 10.000 customers require 60 server of 128 GB RAM

![image](https://user-images.githubusercontent.com/19665800/213906730-13e4a99d-82eb-49c9-a53b-69e62d3a59ef.png)

**Comment by aeneasr** [comment, marked as answer]
Thank you for the details and explanations! , Ory open source does not support multi tenant systems at a scale like the one you describe (and neither do other open source projects like Keycloak). Only our commercial offering (the Ory Network) supports this use case, and we’re happy to talk and figure out how to make it work for you both in terms of economics and scale! Just reach out to support@ory.sh
